feat(trace): Update Trace item details to handle spans

wmak opened the pull request:
- This updates the trace-item-details endpoint to allow item_type=spans
- This updates the list of span attributes so we can get all the attributes correctly
- This fixes a bug where we were checking the python `type` builtin compared to a string instead of `val_type`
- This removes organization_id and item_id from the response since these aren't user facing fields

wmak committed:
feat(trace): Update Trace item details to handle spans

- This updates the trace-item-details endpoint to allow item_type=spans
- This updates the list of span attributes so we can get all the
  attributes correctly
- This fixes a bug where we were checking the python `type` builtin
  compared to a string instead of `val_type`
- This removes organization_id and item_id from the response since these
  aren't user facing fields

wmak committed:
remove: SUPPORTED_TRACE_TYPES

k-fish commented on src/sentry/api/endpoints/project_trace_item_details.py:
```diff
@@ -24,6 +24,17 @@
+}
+
+
+DISALLOW_LIST = {
+    "sentry.organization_id",
+    "sentry.item_type",
+}
```
Not for this PR but:

I think it would be better that we fully map all 'known' attributes in `OURLOG_ATTRIBUTE_DEFINITIONS` regardless of whether we choose to expose them or not. In some way we can mark the `ResolvedAttribute` (or ResolvedColumn) as a column that it not meant to be exposed to the frontend (eg. internal etc.)

Then we can simply convert similar to using the `translate_internal_to_public_alias` for any TraceItemAttribute.

wmak replied:
As discussed, I agree but would prefer to get this change in for now and follow-up with that 👍

k-fish reviewed (commented):
Blocked on https://github.com/getsentry/sentry/pull/88500

wmak committed:
fix: tests

wmak committed:
Merge branch 'master' into wmak/feat/update-trace-item-details-to-handle-spans

wmak committed:
Merge branch 'master' into wmak/feat/update-trace-item-details-to-handle-spans

wmak committed:
fix: move project_id to a common attribute so logs resolves it to
`project_id` and not `sentry.project_id`

wmak committed:
fix: revert change to precise timestamps

- Going to revisit this after, we need to flip dynamically between these
  columns depending on the event timestamp because we only want the new
  field name when we're hitting eap_items.

k-fish committed:
Merge remote-tracking branch 'origin/master' into wmak/feat/update-trace-item-details-to-handle-spans
